src/cmd/linuxkit/pkg_build.go

```diff
@@ -102,9 +130,38 @@ func pkgBuild(args []string) {
 		os.Exit(1)
 	}
 	opts = append(opts, pkglib.WithBuildBuilders(buildersMap))
-	if err := p.Build(opts...); err != nil {
-		fmt.Fprintf(os.Stderr, "%v\n", err)
-		os.Exit(1)
+
+	for _, p := range pkgs {
+		// unless overridden, platforms are specific to a package, so this needs to be inside the for loop
+		if len(plats) == 0 {
+			for _, a := range p.Arches() {
+				if _, ok := skipPlatformsMap[a]; ok {
+					continue
+				}
+				plats = append(plats, imagespec.Platform{OS: "linux", Architecture: a})
+			}
+		}
+		opts = append(opts, pkglib.WithBuildPlatforms(plats...))
```

Comment:
Same here with opts as the variable exists outside this for loop.

Reply:
Ditto.